.github/workflows/codespell.yml

```diff
@@ -0,0 +1,25 @@
+# Codespell configuration is within pyproject.toml
+---
+name: Codespell
```

Comment:
Is there a strong reason for this to be its own workflow rather than folded into `ci.yaml`?

Reply:
could be done so if you like but

- you have already lots of commands in there sequentially and without separate steps -- so one fails (e.g. black) and others already don't run, and here you just see that CI failed but do not know what check exactly
- why don't you use pre-commit and instead list individual calls there? (related: fresh https://github.com/childmindresearch/bids2table/pull/40).  Could be done via action or there is even pre-commit service IIRC (I typically do not use it but other projects do)
- having separate workflow or parametrization for it shows up as a separate report here in PR so makes it easier to immediately see what is wrong. They also then run in parallel

With that in mind I would have recommended to move "pre-commit" encoded checks into separate workflow, and leave/rename `ci` to run only tests but across different python versions (now just 3.8)

You could also encode them all in a single CI workflow file, but then I would also recommend to centralize them actually outside, e.g. in `tox.ini` and then just sweep through different invocations. Eg. look at  https://github.com/con/duct/blob/main/.github/workflows/test.yaml which sweeps through different pythons for testing but also adds runs for  type checks and linting.

so -- many ways to skin a cat here ;-)